client/packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/allocation/useAllocationContext.ts

```diff
@@ -0,0 +1,190 @@
+  isDirty: false,
+  initialisedForItemId: null,
+  draftLines: [],
+  nonAllocatableLines: [],
+  placeholderQuantity: null,
+  alerts: [],
+  allocateIn: AllocateIn.Units,
+
+  initialise: (
+    { item, draftLines, placeholderQuantity },
+    strategy,
+    allowPlaceholder,
+    scannedBatch?: string
+  ) => {
+    const sortedLines = draftLines.sort(SorterByStrategy[strategy]);
+
+    // Separate lines here, so only dealing with allocatable lines going forward
+    // Note - expired is still considered allocatable, just not via auto-allocation
+    const [allocatableLines, nonAllocatableLines] = ArrayUtils.partition(
+      sortedLines,
+      line =>
+        scannedBatch
+          ? scannedBatchFilter(sortedLines, line, scannedBatch)
+          : canAllocate(line)
+    );
+
+    set({
+      isDirty: false,
+      initialisedForItemId: item.id,
+
+      draftLines: allocatableLines,
+      nonAllocatableLines,
+
+      placeholderQuantity: allowPlaceholder ? (placeholderQuantity ?? 0) : null,
+      alerts: [],
+    });
+  },
+
+  clear: () =>
+    set(state => ({
+      ...state,
+      isDirty: false,
+      draftLines: [],
+      nonAllocatableLines: [],
+      placeholderQuantity: null,
+      initialisedForItemId: null,
+      allocateIn: AllocateIn.Units,
+      alerts: [],
+    })),
+
+  setDraftLines: lines =>
+    set(state => ({
+      ...state,
+      isDirty: true,
+      draftLines: lines,
+    })),
+
+  setAlerts: alerts =>
+    set(state => ({
+      ...state,
+      alerts,
+    })),
+
+  autoAllocate: (quantity, format, t) => {
+    const {
+      draftLines,
+      nonAllocatableLines,
+      placeholderQuantity,
+      setDraftLines,
+    } = get();
+
+    const result = allocateQuantities(draftLines, quantity);
+
+    if (result) {
+      setDraftLines(result.allocatedLines);
+
+      const allocatedUnits = getAllocatedUnits({
+        draftLines: result.allocatedLines,
+        placeholderQuantity: 0, // don't want to include any placeholder in this calc
+      });
+
+      const hasOnHold = nonAllocatableLines.some(
+        ({ availablePacks, stockLineOnHold }) =>
+          availablePacks > 0 && !!stockLineOnHold
+      );
+      const hasExpired = draftLines.some(
+        ({ expiryDate }) =>
+          !!expiryDate && DateUtils.isExpired(new Date(expiryDate))
+      );
+
+      const stillToAllocate =
+        result.remainingQuantity > 0 ? result.remainingQuantity : 0;
+
+      const alerts = getAllocationAlerts(
+        quantity,
+        allocatedUnits,
+        stillToAllocate,
+        hasOnHold,
+        hasExpired,
+        format,
+        t
+      );
+
+      set(state => ({
+        ...state,
+        alerts,
+        placeholderQuantity:
+          placeholderQuantity === null ? null : stillToAllocate,
+      }));
+    }
+  },
+
+  manualAllocate: (lineId, quantity) => {
+    const { draftLines, setDraftLines } = get();
+
+    const updatedLines = issueStock(draftLines, lineId, quantity);
```

Comment:
With this need allocateIn (or is that in future PR)

Reply:
Yah future PR